### PR TITLE
build(node_library): migrate to Trampoline V2

### PR DESCRIPTION
**Pull request by tmatsuo**
Tested at:
https://github.com/googleapis/release-please/pull/553
https://github.com/googleapis/nodejs-phishing-protection/pull/200

### CODE REVIEW

File: synthtool/gcp/templates/node_library/.kokoro/trampoline_v2.sh [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+# You can optionally change these environment variables:
+# TRAMPOLINE_IMAGE_UPLOAD:
+#     (true|false): Whether to upload the Docker image after the
+#                   successful builds.
+# TRAMPOLINE_BUILD_FILE: The script to run in the docker container.
+# TRAMPOLINE_WORKSPACE: The workspace path in the docker container.
+#                       Defaults to /workspace.
+# Potentially there are some repo specific envvars in .trampolinerc in
+# the project root.
+
```
Comment (SurferJeffAtGoogle) [inline synthtool/gcp/templates/node_library/.kokoro/trampoline_v2.sh:48]: Please add:
1. A example of a call the this script.
2. A description of how it fits into the larger pipeline.  Does an automated tool call this script?  Or a human?  When?  Who consumers the output docker file?

File: synthtool/gcp/templates/node_library/.kokoro/trampoline.sh [hunk 2]
```diff
@@ -1,28 +0,0 @@
-#!/bin/bash
```
Comment (SurferJeffAtGoogle) [inline synthtool/gcp/templates/node_library/.kokoro/trampoline.sh:1]: To make this PR easier to rollback, please don't delete this file yet.  Please add a comment describing that it's obsolete and add a TODO() to delete it.

File: synthtool/gcp/templates/node_library/.kokoro/populate-secrets.sh [hunk 3]
```diff
@@ -19,25 +19,55 @@ function now { date +"%Y-%m-%d %H:%M:%S" | tr -d '\n' ;}
 function msg { println "$*" >&2 ;}
 function println { printf '%s\n' "$(now) $*" ;}
 
+# Populates requested secrets set in SECRET_MANAGER_KEYS
```
Comment (SurferJeffAtGoogle) [inline synthtool/gcp/templates/node_library/.kokoro/populate-secrets.sh:22]: Who do we expect to call this script?  When?  Why?  Please add comments around line 15.

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]

**Review by bcoe — APPROVED** [review]
This looks good to me, and has been tested here:

https://github.com/googleapis/nodejs-phishing-protection/pull/202

I would like to do an end to end test which actually exercises the secrets, once @SurferJeffAtGoogle's comments are addressed.

File: synthtool/gcp/templates/node_library/.kokoro/trampoline_v2.sh (hunk 1 above)
Comment (tmatsuo) [reply synthtool/gcp/templates/node_library/.kokoro/trampoline_v2.sh:48]: Done

File: synthtool/gcp/templates/node_library/.kokoro/populate-secrets.sh (hunk 3 above)
Comment (tmatsuo) [reply synthtool/gcp/templates/node_library/.kokoro/populate-secrets.sh:22]: Comment added

**Comment by tmatsuo** [comment]
@SurferJeffAtGoogle Thanks for your review. PTAL

File: synthtool/gcp/templates/node_library/.kokoro/populate-secrets.sh (hunk 3 above)
Comment (SurferJeffAtGoogle) [reply synthtool/gcp/templates/node_library/.kokoro/populate-secrets.sh:25]: Looks good.

**Review by SurferJeffAtGoogle — APPROVED** [review]

**Comment by tmatsuo** [comment]
@bcoe Now it's approved. Let me know how the e2e test goes.

**Comment by bcoe** [comment]
@tmatsuo I will do an end to end test Monday 👌 (_as I don't want to release a library on Friday).

**Comment by bcoe** [comment]
@tmatsuo sorry for the delay, my PRs got a bit away from me this week. I have tested this branch, and have good news and bad news.

trampoline 2.0 is mostly working, in that s publish happened:

npm:

https://www.npmjs.com/package/@google-cloud/phishing-protection

kokoro:

https://fusion.corp.google.com/projectanalysis/summary/KOKORO/prod:cloud-devrel%2Fclient-libraries%2Fnodejs%2Frelease%2Fgoogleapis%2Fnodejs-phishing-protection%2Fpublish

However, the final step never happened which updates tags, the command can be found here:

https://github.com/googleapis/nodejs-phishing-protection/blob/master/.kokoro/publish.sh#L23

It received the error:

"No github token or PR specified to report status to, returning."

**Comment by bcoe** [comment]
Is there a chance that AUTORELEASE_PR would not be set with trampoline 2.0?

https://github.com/googleapis/releasetool/blob/ff891a41001f95ed10046adf252bad3bb1b289a8/releasetool/__main__.py#L158

**Comment by tmatsuo** [comment]
@bcoe Maybe. Who set the AUTORELEASE_PR in the first place?

**Comment by tmatsuo** [comment]
@bcoe I see AUTORELEASE_PR is extracted from Kokoro request. I think we need to pass down this envvar into the container.

**Comment by tmatsuo** [comment]
@bcoe I think the last commit should solve the issue. I'll appreciate it if you can try the release again.

**Pull request merged**